tabbycat/api/fields.py

```diff
@@ -0,0 +1,21 @@
+from rest_framework.relations import HyperlinkedIdentityField
+from rest_framework.reverse import reverse
+
+
+class TournamentHyperlinkedIdentityField(HyperlinkedIdentityField):
+
+    def get_url(self, obj, view_name, request, format):
+        lookup_value = getattr(obj, self.lookup_field)
+        kwargs = {
+            'tournament_slug': obj.tournament.slug,
+            self.lookup_url_kwarg: lookup_value,
+        }
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+    def get_object(self, view_name, view_args, view_kwargs):
```

Comment:
What would you think of moving the tournament filter to the `get_queryset` method like with the `SingleObjectFromTournamentMixin`? I think it would be nicer.

Reply:
Hmm, it'd be nice, but then how would you access the relevant `Tournament` instance?

Reply:
I see